app/src/main/java/com/keylesspalace/tusky/components/search/fragments/SearchStatusesFragment.kt

```diff
@@ -0,0 +1,147 @@
+    }
+
+    override fun onReply(position: Int) {
+        (adapter as? SearchStatusesAdapter)?.getItem(position)?.first?.let { status ->
+            reply(status)
+        }
+    }
+
+    override fun onFavourite(favourite: Boolean, position: Int) {
+        //TODO
+    }
+
+    override fun onMore(view: View, position: Int) {
+        (adapter as? SearchStatusesAdapter)?.getItem(position)?.first?.let {
+            more(it, view, position)
+        }
+    }
+
+    override fun onViewMedia(position: Int, attachmentIndex: Int, view: View?) {
+        (adapter as? SearchStatusesAdapter)?.getItem(position)?.first?.actionableStatus?.let { actionable ->
+            when (actionable.attachments[attachmentIndex].type) {
+                Attachment.Type.GIFV, Attachment.Type.VIDEO, Attachment.Type.IMAGE -> {
+                    val attachments = AttachmentViewData.list(actionable)
+                    val intent = ViewMediaActivity.newIntent(context, attachments,
+                            attachmentIndex)
+                    if (view != null) {
+                        val url = actionable.attachments[attachmentIndex].url
+                        ViewCompat.setTransitionName(view, url)
+                        val options = ActivityOptionsCompat.makeSceneTransitionAnimation(requireActivity(),
+                                view, url)
+                        startActivity(intent, options.toBundle())
+                    } else {
+                        startActivity(intent)
+                    }
+                }
+                Attachment.Type.UNKNOWN -> {
+                }
+            }
+
+        }
+
+    }
+
+    override fun onViewThread(position: Int) {
+        (adapter as? SearchStatusesAdapter)?.getItem(position)?.first?.let {
+            viewThread(it)
+        }
+    }
+
+    override fun onOpenReblog(position: Int) {
+        (adapter as? SearchStatusesAdapter)?.getItem(position)?.first?.let {
+            openReblog(it)
+        }
+    }
+
+    override fun onExpandedChange(expanded: Boolean, position: Int) {
+        //TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+    }
+
+    override fun onLoadMore(position: Int) {
+        //Ignore
+    }
+
+    override fun onContentCollapsedChange(isCollapsed: Boolean, position: Int) {
+        //TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+    }
+
+    override fun onVoteInPoll(position: Int, choices: MutableList<Int>) {
+        //TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+    }
+
+    override fun removeItem(position: Int) {
+        //TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+    }
+
+    override fun onReblog(reblog: Boolean, position: Int) {
+        //TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
```

Comment:
Don't forget to implement these methods

Reply:
Implemented. I decided to do a trick.
ViewModel stores loaded statuses at the separate list.
When user update the status - app updates the status at this list and invalidate DataSource. DataSource uses this precached list as initial load data. When user changes the search phrase, app clear this list and app uses the network request to load data.
Not sure that this is the best solution, but it allow to use standard paging libraries.